cognitoclient/mock/cognito.go

```diff
@@ -0,0 +1,24 @@
+package mock
+
+import (
+	"errors"
+
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider/cognitoidentityprovideriface"
+)
+
+type MockCognitoIdentityProviderClient struct {
+    cognitoidentityprovideriface.CognitoIdentityProviderAPI
+}
+
+// mocked functions
+
+func (m *MockCognitoIdentityProviderClient) DescribeUserPool(poolInputData *cognitoidentityprovider.DescribeUserPoolInput) (*cognitoidentityprovider.DescribeUserPoolOutput, error) {
+	exsitingPoolID := "us-west-2_aaaaaaaaa"
+    if *poolInputData.UserPoolId != exsitingPoolID {
+		return nil, errors.New("Failed to load user pool data")
+	} 
+	return nil, nil
+}
+
```

Comment:
This is fine but it's as it's hardcoded it's not flexible (I thinking of future uses where we might need to test different scenarios). If you updated it to something like:
```
type MockCognitoIdentityProviderClient struct {
	cognitoidentityprovideriface.CognitoIdentityProviderAPI

	DescribeUserPoolFunc func(poolInputData *cognitoidentityprovider.DescribeUserPoolInput) (*cognitoidentityprovider.DescribeUserPoolOutput, error)
}

func (m *MockCognitoIdentityProviderClient) DescribeUserPool(poolInputData *cognitoidentityprovider.DescribeUserPoolInput) (*cognitoidentityprovider.DescribeUserPoolOutput, error) {
	return m.DescribeUserPoolFunc()
}
```
It can be customised to behave however is required. When you construct the mock you give it the function you want it to execute.
````
m := &MockCognitoIdentityProviderClient{
		DescribeUserPoolFunc := func(poolInputData *cognitoidentityprovider.DescribeUserPoolInput) (*cognitoidentityprovider.DescribeUserPoolOutput, error) {
			// set behaviour as reqiured here.
		}
	}
````
Not strictly necessary but food for thought.